AxonDeepSeg/segment.py

```diff
@@ -108,10 +139,38 @@ def segment_image(
 
         img_name_original = acquisition_name.stem
 
+        # Get the model's native resolution
+        model_resolution, patch_size = get_model_native_resolution_and_patch(path_model)
+
+        # Check that the resampled image will be of sufficient size, and if not throw an error.
+        im = Image.open(path_testing_image)
+        w, h = im.size
+
+        w_resampled = w*(acquired_resolution*zoom_factor)/model_resolution
+        h_resampled = h*(acquired_resolution*zoom_factor)/model_resolution
+
+        if w_resampled < patch_size or h_resampled < patch_size:
+            if w<=h:
+                minimum_zoom_factor = patch_size*model_resolution/(w*acquired_resolution)
+            else:
+                minimum_zoom_factor = patch_size*model_resolution/(h*acquired_resolution)
+
+            # Round to 1 decimal, always up.
+            minimum_zoom_factor = ceil(minimum_zoom_factor*10)/10
+
+            print("ERROR: Due to your given image size, resolution, and zoom factor, the resampled image is smaller than",
+                   "the patch size during segmentation. To resolve this, please set a zoom factor greater than ",
+                   str(minimum_zoom_factor), ".",
+                   "To do this on the command line, call the segmentation with the -z flag, i.e. ",
+                   "-z ", str(minimum_zoom_factor),
+            )
+            sys.exit(4)
```

Comment:
The `sys.exit` makes `fsleyes` quit. Is there a way to display to avoid that and still have the error in the fsleyes terminal?

Reply:
Hum, I used this because we were already using this elsewhere already, see here for an example: 

https://github.com/axondeepseg/axondeepseg/blob/ca4baad96bed0d9c84b2dc63a8c4b464dc4d5a2c/AxonDeepSeg/segment.py#L268-L272

I guess there, it's only called when the CLI is called. Should we handle the zoom factor error there as well, and then in FSLeyes seperately in the plugin?

Reply:
We can wrap the call to segment_image in the plugin around a try-catch block and intercept the `SystemExit` exception.

Reply:
Ahh good idea, and we could in fact trigger the FSLeyes popup with the message if the sys error code 4 is triggered.

Reply:
@mathieuboudreau yep that's what I had in mind. I can take care of it (for some reason fsleyes suddently works on my linux)

Reply:
Only problem is that the minimum zoom factor value is lost

Reply:
Ah but we can tell the user to take a look at his terminal to check the minimum zoom factor because the "error" message doesn't actually comes from an error so it is still printed.

Reply:
Ok so this is now fixed.

![Screenshot_20220323_173148](https://user-images.githubusercontent.com/83031821/159799320-a9d5cd8d-00c7-459f-8d15-0a0cbd2c7f3d.png)